src/main/java/graphql/schema/idl/WiringFactory.java

```diff
@@ -53,18 +54,50 @@
      * This is called to ask if this factory can provide a data fetcher for the definition
      *
      * @param registry   the registry of all types
+     * @param parentType the type of the parent
      * @param definition the field definition in play
      * @return true if the factory can give out a date fetcher
      */
-    boolean providesDataFetcher(TypeDefinitionRegistry registry, FieldDefinition definition);
+    default boolean providesDataFetcher(TypeDefinitionRegistry registry, TypeDefinition parentType, FieldDefinition definition) {
```

Comment:
No, we don't really need to preserve the API here: It is a very new feature and I don't expect it to be used for the "standard" use-case.

The annotation was missing, but to be honest, we didn't clearly communicate it yet, what the annotations means and what to expect. This is still a todo. 

And yes: we should refactor all parameters to take an Parameter-Object to make it safe to expand it in the future.